### ISSUE TITLE
Assembly translation failures for zstd and libxml2

### ISSUE THREAD

**Issue by thedataking** [immunant/c2rust#62]
Translating [zstd](https://github.com:immunant/zstd) like this (working around other issues):

    intercept-build make -j`ncpus` zstd
    c2rust transpile --translate-asm -m zstdcli compile_commands.json
    cargo build

produces:

```
error: couldn't allocate output register for constraint 'a'
    --> programs/../lib/compress/zstd_compress.rs:1778:5
     |
1778 |     asm!("cpuid" : "=a" (n) : "a" (0i32) : "ebx" , "ecx" , "edx");
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: couldn't allocate output register for constraint 'a'
    --> programs/../lib/compress/zstd_compress.rs:1781:9
     |
1781 | /         asm!("cpuid" : "=a" (f1a) , "=c" (f1c) , "=d" (f1d) : "a" (1i32) :
1782 | |              "ebx")
     | |___________________^

error: couldn't allocate output register for constraint 'a'
    --> programs/../lib/compress/zstd_compress.rs:1786:9
     |
1786 | /         asm!("cpuid" : "=a" (f7a) , "=b" (f7b) , "=c" (f7c) : "a" (7i32) , "c"
1787 | |              (0i32) : "edx")
     | |____________________________^

error: couldn't allocate output register for constraint 'a'
    --> programs/../lib/decompress/zstd_decompress.rs:1050:5
     |
1050 |     asm!("cpuid" : "=a" (n) : "a" (0i32) : "ebx" , "ecx" , "edx");
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: couldn't allocate output register for constraint 'a'
    --> programs/../lib/decompress/zstd_decompress.rs:1053:9
     |
1053 | /         asm!("cpuid" : "=a" (f1a) , "=c" (f1c) , "=d" (f1d) : "a" (1i32) :
1054 | |              "ebx")
     | |___________________^

error: couldn't allocate output register for constraint 'a'
    --> programs/../lib/decompress/zstd_decompress.rs:1058:9
     |
1058 | /         asm!("cpuid" : "=a" (f7a) , "=b" (f7b) , "=c" (f7c) : "a" (7i32) , "c"
1059 | |              (0i32) : "edx")
     | |____________________________^
```

The assembly code originates from https://github.com/immunant/zstd/blob/dev/lib/common/cpu.h#L34-L110. Problem persists whether one builds with `clang` or `gcc`.

**Comment by thedataking** [comment]
Translating libxml2 (on `feature/va_copy` + custom toolchain with va_copy support) produces similar errors:

```
error: couldn't allocate output register for constraint 'c'
    --> nanoftp.rs:2169:5
     |
2169 | /     asm!("cld; rep; stosq" : "=c" (__d0) , "=D" (__d1) : "a" (0i32) , "0"
2170 | |          ((::std::mem::size_of::<fd_set>() as
2171 | |       libc::c_ulong).wrapping_div(::std::mem::size_of::<__fd_mask>() as
2172 | |                                       libc::c_ulong))
2173 | |          , "1"
2174 | |          (&mut *rfd.__fds_bits.as_mut_ptr().offset(0isize) as *mut __fd_mask)
2175 | |          : "memory" : "volatile");
     | |__________________________________^

error: couldn't allocate output register for constraint 'c'
    --> nanoftp.rs:2470:5
     |
2470 | /     asm!("cld; rep; stosq" : "=c" (__d0) , "=D" (__d1) : "a" (0i32) , "0"
2471 | |          ((::std::mem::size_of::<fd_set>() as
2472 | |       libc::c_ulong).wrapping_div(::std::mem::size_of::<__fd_mask>() as
2473 | |                                       libc::c_ulong))
2474 | |          , "1"
2475 | |          (&mut *rfd.__fds_bits.as_mut_ptr().offset(0isize) as *mut __fd_mask)
2476 | |          : "memory" : "volatile");
     | |__________________________________^

error: couldn't allocate output register for constraint 'c'
    --> nanoftp.rs:2488:5
     |
2488 | /     asm!("cld; rep; stosq" : "=c" (__d0_0) , "=D" (__d1_0) : "a" (0i32) , "0"
2489 | |          ((::std::mem::size_of::<fd_set>() as
2490 | |       libc::c_ulong).wrapping_div(::std::mem::size_of::<__fd_mask>() as
2491 | |                                       libc::c_ulong))
2492 | |          , "1"
2493 | |          (&mut *efd.__fds_bits.as_mut_ptr().offset(0isize) as *mut __fd_mask)
2494 | |          : "memory" : "volatile");
     | |__________________________________^

error: couldn't allocate output register for constraint 'c'
    --> nanoftp.rs:3394:9
     |
3394 | /         asm!("cld; rep; stosq" : "=c" (__d0) , "=D" (__d1) : "a" (0i32) , "0"
3395 | |              ((::std::mem::size_of::<fd_set>() as
3396 | |       libc::c_ulong).wrapping_div(::std::mem::size_of::<__fd_mask>() as
3397 | |                                       libc::c_ulong))
3398 | |              , "1"
3399 | |              (&mut *rfd.__fds_bits.as_mut_ptr().offset(0isize) as *mut __fd_mask)
3400 | |              : "memory" : "volatile");
     | |______________________________________^

error: couldn't allocate output register for constraint 'c'
    --> nanoftp.rs:3881:9
     |
3881 | /         asm!("cld; rep; stosq" : "=c" (__d0) , "=D" (__d1) : "a" (0i32) , "0"
3882 | |              ((::std::mem::size_of::<fd_set>() as
3883 | |       libc::c_ulong).wrapping_div(::std::mem::size_of::<__fd_mask>() as
3884 | |                                       libc::c_ulong))
3885 | |              , "1"
3886 | |              (&mut *rfd.__fds_bits.as_mut_ptr().offset(0isize) as *mut __fd_mask)
3887 | |              : "memory" : "volatile");
     | |______________________________________^

error: couldn't allocate output register for constraint 'c'
    --> nanoftp.rs:3900:9
     |
3900 | /         asm!("cld; rep; stosq" : "=c" (__d0_0) , "=D" (__d1_0) : "a" (0i32) ,
3901 | |              "0"
3902 | |              ((::std::mem::size_of::<fd_set>() as
3903 | |       libc::c_ulong).wrapping_div(::std::mem::size_of::<__fd_mask>() as
...    |
3906 | |              (&mut *efd.__fds_bits.as_mut_ptr().offset(0isize) as *mut __fd_mask)
3907 | |              : "memory" : "volatile");
     | |______________________________________^
```

**Comment by rinon** [comment]
Should be fixed in 44d3eb5
